R/prophet.R

```diff
@@ -277,8 +312,87 @@ do_prophet_ <- function(df, time_col, value_col = NULL, periods, time_unit = "da
   # thanks to avoid_conflict that is used before,
   # this doesn't overwrite grouping columns.
   tmp_col <- avoid_conflict(colnames(df), "tmp_col")
-  test <- df %>%
+  ret <- df %>%
     dplyr::do_(.dots=setNames(list(~do_prophet_each(.)), tmp_col)) %>%
     dplyr::ungroup()
-  test %>%  unnest_with_drop_(tmp_col)
+  ret <- ret %>%  unnest_with_drop_(tmp_col)
+
+  if (length(grouped_col) > 0) {
+    ret <- ret %>% dplyr::group_by(!!!rlang::syms(grouped_col))
+  }
+  ret
+}
+
+
+#' Calculate MAE.
+#' @param actual - Actual value.
+#' @param predicted - Predicted value.
+#' @param is_test_data - logical vector that indicates test data portion of actual and predicted.
+#' @export
```

Comment:
I think those functions are generic so those should be defined in more generic place like system.R. If I'm wrong, please le t me know.

Reply:
Moved those functions to util.R.